app/models/category.rb

```diff
@@ -2,8 +2,90 @@ class Category < ActiveRecord::Base
   belongs_to :site
   has_many :posts
 
+  has_ancestry
+  acts_as_list scope: [:ancestry], column: :order
+
   validates :slug, format: /\A\w+\z/, uniqueness: {scope: :site_id}
-  validates :order, numericality: :only_integer
+  validates :order, numericality: {only_integer: true, allow_nil: true}
 
   scope :ordered, -> { order(:order) }
+  # For PostgreSQL, doesn't work on MySQL
+  # See https://github.com/stefankroes/ancestry/pull/238
+  scope :leaves, -> { joins("LEFT JOIN #{quoted_table_name} AS c ON c.#{ancestry_column} = CAST(#{quoted_table_name}.id AS text) OR c.#{ancestry_column} = #{quoted_table_name}.#{ancestry_column} || '/' || #{quoted_table_name}.id").group("#{quoted_table_name}.id").having("COUNT(c.id) = 0").order("#{quoted_table_name}.#{ancestry_column} NULLS FIRST").order(:order) }
+  scope :parental, ->(category = nil) { includes(:posts).where("posts.id" => nil).where.not(id: category&.id).order("#{quoted_table_name}.#{ancestry_column} NULLS FIRST").order(:order) }
```

Comment:
`order("#{quoted_table_name}.#{ancestry_column} NULLS FIRST").order(:order)` is used twice.
Can you extract it as scope?